numcodecs/vlen.pyx

```diff
@@ -74,7 +75,7 @@ class VLenUTF8(Codec):
     def encode(self, buf):
         cdef:
             Py_ssize_t i, l, n_items, data_length, total_length
-            object[:] input_values
+            ndarray[object, ndim=1] input_values
```

Comment:
Isaac, could you please share a bit more about why this is needed?

Reply:
I don't remember that much of the specifics, but it has to do with which types Cython is able to work with a read-only buffer. I remember running into this a bit with pandas too when I tried to make the index arrays actually read-only.

Is that what you were wondering about?

Reply:
More info here: https://github.com/zarr-developers/numcodecs/issues/514#issuecomment-1992070983

Reply:
Yes thank you! 🙏